pkg/cmd/roachtest/tests/versionupgrade.go

```diff
@@ -287,3 +290,69 @@ for i in 1 2 3 4; do
 done
 `)
 }
+
+func registerHTTPRestart(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:             "http-restart/mixed-version",
+		Owner:            registry.OwnerTestEng,
+		Cluster:          r.MakeClusterSpec(4),
+		CompatibleClouds: registry.AllClouds,
+		Suites:           registry.Suites(registry.MixedVersion, registry.Nightly),
+		Randomized:       true,
+		Run:              runHTTPRestart,
+		Timeout:          1 * time.Hour,
+	})
+}
+
+func runHTTPRestart(ctx context.Context, t test.Test, c cluster.Cluster) {
+	mvt := mixedversion.NewTest(ctx, t, t.L(), c,
+		c.CRDBNodes(),
+		mixedversion.MinimumSupportedVersion("v24.2.0"),
+		mixedversion.MinUpgrades(2),
+		mixedversion.EnabledDeploymentModes(mixedversion.SeparateProcessDeployment),
+	)
+
+	httpCall := func(ctx context.Context, node int, l *logger.Logger) {
+		logEvery := roachtestutil.Every(1 * time.Second)
+		adminURLs, err := c.ExternalAdminUIAddr(ctx, l, c.Node(node))
+		if err != nil {
+			t.Fatal(err)
+		}
+		url := "https://" + adminURLs[0] + "/ts/query"
+		l.Printf("Sending requests to %s", url)
+		client := roachtestutil.DefaultHTTPClient(c, l)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			// Copied from elsewhere, actual HTTP req shouldn't matter
```

Comment:
Interesting, I tried changing this to a simpler request (`/_status/vars`) and couldn't get it to fail after stressing 10 times. Is a tsdb request much faster so we hit the race condition (much) more often?

Reply:
@DarrylWong You were just unlucky in selecting `/_status/vars` as it [doesn't use the authenticated handler](https://github.com/cockroachdb/cockroach/blob/c68c559859be738efead9971f5e11f62a8c69d06/pkg/server/server_http.go#L216-L217).
TSDB reqs don't require any sort of sql role but they do go through the auth http handler.